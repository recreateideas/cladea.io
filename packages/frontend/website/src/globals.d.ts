import { Theme } from "./App/ui-core/theme";
import { Store } from "redux";

export declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: any;
    /** enzyme global methods for testing */
    shallow: any;
    render: any;
    mount: any;
    theme: Theme;
    ResizeObserver: any;
    store: Store;
  }
}

declare module "ui-core";
