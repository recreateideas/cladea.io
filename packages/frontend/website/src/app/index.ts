export * from "./App";
export * from "./features";
