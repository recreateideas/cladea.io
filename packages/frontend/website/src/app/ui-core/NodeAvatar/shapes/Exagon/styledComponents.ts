import styled, { css } from 'styled-components';

interface IExagon {
    inner?: boolean;
    backgroundColor: string;
    width: number;
    height: number;
    border: number;
}

const exagon = ({ inner, width, height, border, backgroundColor }: IExagon) => {
    return css`
        ${inner &&
        css`
            left: 1px;
            z-index: 1;
        `}
        position: relative;
        height: ${height}px;
        width: ${width}px;
        background-color: ${backgroundColor};
        margin: ${height / 2}px 0;
        &:before,
        &:after {
            content: '';
            position: absolute;
            width: 0;
            border-left: ${width / 2}px solid transparent;
            border-right: ${width / 2}px solid transparent;
        }
        &:before {
            bottom: 100%;
            border-bottom: ${border}px solid ${backgroundColor};
        }
        &:after {
            top: 100%;
            width: 0;
            border-top: ${border}px solid ${backgroundColor};
        }
    `;
};

export const Container = styled.div<{ size?: 'small' | 'medium'; inner?: boolean }>`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    palette,
                    // layout,
                },
            },
            inner,
            size = 'medium',
        } = props;
        const sizes = {
            small: 62,
            medium: 102,
        };
        const length = sizes[size];
        const whRatio = 1.732;
        const width = inner ? length - 2 : length;
        const height = inner ? width / whRatio + 1 : width / whRatio;
        const border = inner ? height / 2 - 1 : height / 2;
        const backgroundColor = inner ? palette.neutrals.secondaryBg : palette.neutrals.borders;
        return css`
            ${exagon({ inner, width, height, border, backgroundColor })}
            .content {
                position: relative;
                top: -1px;
                .icon {
                    display: flex;
                    justify-content: center;
                }
                .header {
                    flex-direction: column;
                    text-align: center;
                }
                .react-flow__handle-bottom {
                    top: ${height +
                    border -
                    2 +
                    1}px; // 1px because content is pushed up 1px in line 69
                    z-index: 2;
                }
            }
        `;
    }}
`;
