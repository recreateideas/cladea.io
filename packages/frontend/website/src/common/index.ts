export * from './utils';
export * from './hooks';
export * from './converters';
