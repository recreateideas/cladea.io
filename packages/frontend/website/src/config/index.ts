export const apiGatewaUrl = process.env.REACT_APP_API_GATEWAY_URL;

console.log(process.env);
