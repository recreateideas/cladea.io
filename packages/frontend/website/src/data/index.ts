import land110m from "./land-110m.json";

export { land110m };
