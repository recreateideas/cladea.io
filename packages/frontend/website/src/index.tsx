import "./app/ui-core/theme/GlobalStyle/roboto.css";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { store } from "./redux";
import { ServicedApp, ThemeProviderConnected } from "./app";

ReactDOM.render(
  <Provider store={store}>
    <ThemeProviderConnected>
      <ServicedApp />
    </ThemeProviderConnected>
  </Provider>,
  document.getElementById("root")
);
