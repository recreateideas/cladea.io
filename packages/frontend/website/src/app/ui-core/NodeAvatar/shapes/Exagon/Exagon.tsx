import { ReactElement } from 'react';
import { ThemedStyledProps } from 'styled-components';
import { Container } from './styledComponents';

interface IExagonProps {
    size?: 'small' | 'medium';
    isActive?: boolean;
    customCss?: ThemedStyledProps<any, any>;
    onMouseEnter?: any;
    onMouseLeave?: any;
    children?: ReactElement | Array<ReactElement>;
}
export const Exagon = (props: IExagonProps): ReactElement => {
    const { size = 'medium', customCss, isActive, onMouseEnter, onMouseLeave, children } = props;
    return (
        <Container
            className="exagon"
            {...{ onMouseEnter, onMouseLeave, customCss, isActive, size }}
        >
            <Container className="exagon inner" {...{ inner: true, size }}>
                {children}
            </Container>
        </Container>
    );
};

Exagon.displayName = 'Exagon';
