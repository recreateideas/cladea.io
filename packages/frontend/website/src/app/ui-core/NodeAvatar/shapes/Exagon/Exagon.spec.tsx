import { Exagon } from './Exagon';
import { TestProvider } from 'src/tests';

const { mountWithProviderAndTheme } = TestProvider();
describe('<Exagon/>', () => {
    it('should render', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<Exagon />, state);
        expect(root).toMatchSnapshot();
    });
});
