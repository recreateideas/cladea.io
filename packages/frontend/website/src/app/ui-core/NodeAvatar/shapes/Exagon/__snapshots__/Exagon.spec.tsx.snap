// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Exagon/> should render 1`] = `
Array [
  .c0 {
  position: relative;
  height: 58.891454965357966px;
  width: 102px;
  background-color: #dedeeb;
  margin: 29.445727482678983px 0;
}

.c0:before,
.c0:after {
  content: '';
  position: absolute;
  width: 0;
  border-left: 51px solid transparent;
  border-right: 51px solid transparent;
}

.c0:before {
  bottom: 100%;
  border-bottom: 29.445727482678983px solid #dedeeb;
}

.c0:after {
  top: 100%;
  width: 0;
  border-top: 29.445727482678983px solid #dedeeb;
}

.c0 .content {
  position: relative;
  top: -1px;
}

.c0 .content .icon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .content .header {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
}

.c0 .content .react-flow__handle-bottom {
  top: 87.33718244803694px;
  z-index: 2;
}

.c1 {
  left: 1px;
  z-index: 1;
  position: relative;
  height: 58.736720554272516px;
  width: 100px;
  background-color: #f9fafb;
  margin: 29.368360277136258px 0;
}

.c1:before,
.c1:after {
  content: '';
  position: absolute;
  width: 0;
  border-left: 50px solid transparent;
  border-right: 50px solid transparent;
}

.c1:before {
  bottom: 100%;
  border-bottom: 28.368360277136258px solid #f9fafb;
}

.c1:after {
  top: 100%;
  width: 0;
  border-top: 28.368360277136258px solid #f9fafb;
}

.c1 .content {
  position: relative;
  top: -1px;
}

.c1 .content .icon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 .content .header {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
}

.c1 .content .react-flow__handle-bottom {
  top: 86.10508083140877px;
  z-index: 2;
}

<div
    className="c0 exagon"
    size="medium"
  >
    <div
      className="c1 exagon inner"
      size="medium"
    />
  </div>,
  "",
]
`;
